src/main/webapp/app/exam/participate/exercises/text/text-exam-submission.component.ts

```diff
@@ -43,12 +52,58 @@ export class TextExamSubmissionComponent extends ExamSubmissionComponent impleme
     ngOnInit(): void {
         // show submission answers in UI
         this.updateViewFromSubmission();
+
+        this.subscriptionToLiveExamExerciseUpdates = this.examExerciseUpdateService.currentExerciseIdAndProblemStatement.subscribe((update) => {
+            this.updateExerciseProblemStatementById(update.exerciseId, update.problemStatement);
+        });
+    }
+
+    toggleHighlightedProblemStatement(): void {
+        if (this.showHighlightedDifferences) {
+            this.getExercise().problemStatement = this.updatedProblemStatement;
```

Comment:
Why aren't we just using `this.exercise` instead of `this.getExercise()` here and below? I was confused since I assume `getExercise` might be doing something more than just returning `this.exercise`.

Reply:
Great find, you are correct. 
I fixed the comment and replaced getExercise() with simply this.exercise